File: node/src/templates/angular-templates.ts
const componentTemplate = (componentClassName: string, fileName: string) =>
`import { Component } from '@angular/core';

@Component({
  selector: 'app-${fileName}',
  templateUrl: './${fileName}.component.html',
  styleUrls: ['./${fileName}.component.scss']
})
export class ${componentClassName} {
  // Component logic here
}
`;

const htmlTemplate = (componentClassName: string, fileName: string) =>
`<p>
  ${componentClassName} works!
</p>
`;

const testTemplate = (componentClassName: string, fileName: string) =>
`import { ${componentClassName} } from './${fileName}.component';

describe('${componentClassName} Component', () => {
  test('it should render', () => {
    render(${componentClassName});
    expect(screen.getByText('${componentClassName} component')).toBeInTheDocument();
  });
});
`;

const styleTemplate = (componentClassName: string, fileName: string) =>
`/* Add your styles here */
`;

export const getTemplate = (type: string) => {
  switch(type) {
    case 'component':
      return componentTemplate;
    case 'html':
      return htmlTemplate;
    case 'test':
      return testTemplate;
    case 'style':
      return styleTemplate;
    default:
      return () => 'template';
  }
}

File: python/templates/angular_templates.py
def angular_component_template(component_class_name, file_name):
    return f"""
import {{ Component }} from '@angular/core';

@Component({{
  selector: 'app-{file_name}',
  templateUrl: './{file_name}.component.html',
  styleUrls: ['./{file_name}.component.scss']
}})
export class {component_class_name} {{
  // Component logic here
}}
"""

def angular_html_template(component_class_name):
    return f"<p>{component_class_name} works!</p>"

def angular_test_template(component_class_name, file_name):
    return f"""
import {{ {component_class_name} }} from './{file_name}.component';

describe('{component_class_name} Component', () => {{
  test('it should render', () => {{
    render({component_class_name});
    expect(screen.getByText('{component_class_name} component')).toBeInTheDocument();
  }});
}});
"""

def angular_style_template(component_class_name):
    return "/* Add your styles here */"